gcsio/src/main/java/com/google/cloud/hadoop/gcsio/StorageStubProvider.java

```diff
@@ -158,15 +158,23 @@ private ChannelAndRequestCounter buildManagedChannel() {
         isNullOrEmpty(readOptions.getGrpcServerAddress())
             ? DEFAULT_GCS_GRPC_SERVER_ADDRESS
             : readOptions.getGrpcServerAddress();
-    ManagedChannel channel =
-        (isComputeCredential()
-                ? GoogleDefaultChannelBuilder.forTarget(target)
-                : ManagedChannelBuilder.forTarget(target))
-            .enableRetry()
-            .defaultServiceConfig(getGrpcServiceConfig())
-            .intercept(counter)
-            .userAgent(userAgent)
-            .build();
+    ManagedChannel channel;
+    if (useDirectpath) {
```

Comment:
nit: is it possible to extract the common calls outside `if` so it's clear(er) what's shared between the implementations?